exporter/elasticsearchexporter/model.go

```diff
@@ -41,16 +41,65 @@ const (
 
 func (m *encodeModel) encodeLog(resource pcommon.Resource, record plog.LogRecord, scope pcommon.InstrumentationScope) ([]byte, error) {
 	var document objmodel.Document
-	document.AddTimestamp("@timestamp", record.Timestamp()) // We use @timestamp in order to ensure that we can index if the default data stream logs template is used.
-	document.AddTraceID("TraceId", record.TraceID())
-	document.AddSpanID("SpanId", record.SpanID())
-	document.AddInt("TraceFlags", int64(record.Flags()))
-	document.AddString("SeverityText", record.SeverityText())
-	document.AddInt("SeverityNumber", int64(record.SeverityNumber()))
-	document.AddAttribute("Body", record.Body())
-	m.encodeAttributes(&document, record.Attributes())
-	document.AddAttributes("Resource", resource.Attributes())
-	document.AddAttributes("Scope", scopeToAttributes(scope))
+
+	switch m.mode {
+	case MappingECS:
+		document.AddTimestamp("@timestamp", record.Timestamp()) // We use @timestamp in order to ensure that we can index if the default data stream logs template is used.
+		document.AddTraceID("trace.id", record.TraceID())
+		document.AddSpanID("span.id", record.SpanID())
+
+		if f := record.Flags(); f != 0 {
+			document.AddInt("trace.flags", int64(record.Flags()))
+		}
+
+		if n := record.SeverityNumber(); n != plog.SeverityNumberUnspecified {
+			document.AddInt("log.syslog.severity.code", int64(record.SeverityNumber()))
+			document.AddString("log.syslog.severity.name", record.SeverityText())
```

Comment:
I think this part needs more thinking. ECS also has [`log.level`](https://www.elastic.co/guide/en/ecs/master/ecs-log.html#field-log-level) - if we deal with syslog, then this part would correctly map under `log.syslog.severity.*`, but if it's not syslog, then it should be under `log.level`.